feature-swap-impl/src/main/java/io/novafoundation/nova/feature_swap_impl/presentation/confirmation/SwapConfirmationViewModel.kt

```diff
@@ -1,68 +1,421 @@
+    val addressFlow: Flow<AddressModel> = combine(chainWithAssetFlow, metaAccountFlow) { chainWithAsset, metaAccount ->
+        val accountId = metaAccount.requireAddressIn(chainWithAsset.chain)
+        addressIconGenerator.createAddressModel(chainWithAsset.chain, accountId, 5, accountName = null, background = BACKGROUND_TRANSPARENT)
+    }
+
+    val slippageAlertMessage: Flow<String?> = slippageAlertMixin.slippageAlertMessage
+
+    init {
+        handleMaxClick()
+
+        initConfirmationState()
+    }
+
+    fun backClicked() {
+        swapRouter.back()
     }
 
     fun rateClicked() {
-        TODO("Not yet implemented")
+        launchDescriptionBottomSheet(
+            titleRes = R.string.swap_rate_title,
+            descriptionRes = R.string.swap_rate_description
+        )
     }
 
     fun priceDifferenceClicked() {
-        TODO("Not yet implemented")
+        launchDescriptionBottomSheet(
+            titleRes = R.string.swap_price_difference_title,
+            descriptionRes = R.string.swap_price_difference_description
+        )
     }
 
     fun slippageClicked() {
-        TODO("Not yet implemented")
+        launchDescriptionBottomSheet(
+            titleRes = R.string.swap_slippage_title,
+            descriptionRes = R.string.swap_slippage_description
+        )
     }
 
     fun networkFeeClicked() {
-        TODO("Not yet implemented")
+        launchDescriptionBottomSheet(
+            titleRes = R.string.swap_network_fee_title,
+            descriptionRes = R.string.swap_network_fee_description
+        )
     }
 
     fun accountClicked() {
-        TODO("Not yet implemented")
+        launch {
+            val chainWithAsset = chainWithAssetFlow.first()
+            val addressModel = addressFlow.first()
+
+            val type = ExternalActions.Type.Address(addressModel.address)
+
+            externalActions.showExternalActions(type, chainWithAsset.chain)
+        }
     }
 
     fun confirmButtonClicked() {
-        TODO("Not yet implemented")
+        launch {
+            val validationSystem = swapInteractor.validationSystem()
+            val payload = getValidationPayload() ?: return@launch
+
+            validationExecutor.requireValid(
+                validationSystem = validationSystem,
+                payload = payload,
+                progressConsumer = _validationProgress.progressConsumer(),
+                validationFailureTransformerCustom = ::formatValidationFailure,
+            ) { validPayload ->
+                executeSwap(validPayload.swapExecuteArgs)
+                _validationProgress.value = false
+            }
+        }
+    }
+
+    private fun executeSwap(swapExecuteArgs: SwapExecuteArgs) {
+        launch {
+            val result = withContext(Dispatchers.Default) { swapInteractor.executeSwap(swapExecuteArgs) }
+            result.onSuccess {
+                swapRouter.finishSwapFlow(swapExecuteArgs.assetIn.fullId.toAssetPayload())
+            }
```

Comment:
Maybe it would be nice to add fallback here? What do you think?

I guess it's really rare case when we get a failure here